OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -188,8 +188,10 @@ public final View getView(int position, View convertView, ViewGroup parent) {
                     final Activity activity = parentFragment.getActivity();
                     if (activity != null && ((RoboAppCompatActivity) activity).isInForeground()) {
                         final BulkDownloadFragment fragment = new BulkDownloadFragment(downloadListener, environment);
-                        parentFragment.getChildFragmentManager().
-                                beginTransaction().replace(convertView.getId(), fragment).commit();
+                        final View finalConvertView = convertView;
+                        // Wait until convertView attached with parent view.
+                        convertView.post(() -> parentFragment.getChildFragmentManager().
+                                beginTransaction().replace(finalConvertView.getId(), fragment).commit());
```

Comment:
`.commitNow()` instead of `.commit()` make more sense here.
@miankhalid your thoughts on it.
Fix sounds good to me.

Reply:
> .commitNow() instead of .commit() make more sense here.

how does it make more sense and why?

Reply:
Because 'commit' do the transaction asynchronously and by putting it in runnable we are already doing it asynchronously.